HW1

Skiftare opened the pull request:
Проверки checksyle пройдены

Skiftare committed:
HW1

Skiftare committed:
Create SampleTest.java

Skiftare committed:
Create Main.java

Skiftare committed:
Create EvenArrayUtils.java

Skiftare committed:
Update Task5.java

kkkoltunov commented on src/main/java/edu/hw1/Task1.java:
```diff
@@ -0,0 +1,41 @@
+
+class Task1 {
+    static int minutesToSeconds(@NotNull String s) {
+        int ans = -1;
```
обращай внимание на подсказки ide, стараемся держать в коде как можно меньше ворнингов, тк это плохой тон

данную переменную можно объявить сразу в строке 33
```
 int ans = mins * sixtyNum + secs;
```

kkkoltunov commented on src/main/java/edu/hw1/Task1.java:
```diff
@@ -0,0 +1,41 @@
+import org.jetbrains.annotations.NotNull;
+
+@SuppressWarnings("HideUtilityClassConstructor")
+
```
между аннотациями и объектами над которыми она ставиться не нужно ставить отступы, давай уберем во всех классах)

Skiftare replied:
Сейчас будет

Skiftare committed:
Update Task1.java

Убрал лишние отступы

Skiftare committed:
Update Task1.java

Исправлено объявление  ans

kkkoltunov commented on src/main/java/edu/hw1/Task1.java:
```diff
@@ -0,0 +1,41 @@
+        boolean flag = false;
+        int mins = 0;
+        @SuppressWarnings("MagicNumber")
+        int tenMulti = 10;
```
можно кстати очень просто избегать случаев когда необходимо вешать supress для меджик намберов, например
```
    private static final Integer TEN_MULTI = 10;
```
и потом использовать это в коде, тогда и ворнингов не будет и код будет красивый)

Skiftare committed:
Update Task1.java

Skiftare committed:
Update Task2.java

Убраны отступы между классами

Skiftare committed:
Update Task1.java

Skiftare committed:
Update Task5.java

Избавился от MagicNumber через  private static final Integer

Skiftare committed:
Update Task6.java

Заменил magicnumber на private static final Integer

kkkoltunov commented on src/main/java/edu/hw1/Task1.java:
```diff
@@ -0,0 +1,37 @@
+        private static final Integer tenMulti = 10;
+        private static final Integer sixtyNum = 60;
+
+        for (int i = 0; i < s.length(); i++) {
```
кажется тут немного переусложнил, а что если попробовать: 
- функцией split от строки
- Integer.parseInt
- выполнять дальнейшую обработку чисел

Skiftare replied:
Окей. Решил просто, что строки в числа, долго, неэффективно... Сейчас сделаю так. Наследие спортпрога)

kkkoltunov commented on src/main/java/edu/hw1/Task1.java:
```diff
@@ -0,0 +1,37 @@
+        int secs = 0;
+        boolean flag = false;
+        int mins = 0;
+        private static final Integer tenMulti = 10;
```
давай это вынесем прям в константы класса, и важно сохранять upper case snake case для констант, то есть так 
```
class Task1 {

    private static final Integer TEN_MULTI = 10;

    static int minutesToSeconds(@NotNull String s) {
        int secs = 0;
        boolean flag = false;
        int mins = 0;
```

kkkoltunov commented on src/test/java/edu/hw1/Task2Test.java:
```diff
@@ -0,0 +1,22 @@
+        assertEquals(3, Task2.countDigits(123));
+    }
+
+    void zero() {
```
а почему на этот метод не навесил аннотацию Test и на тест ниже?

kkkoltunov commented on src/main/java/edu/hw1/Task2.java:
```diff
@@ -0,0 +1,22 @@
+public class Task2 {
+
+    @SuppressWarnings("MagicNumber")
+    public static int countDigits(int n) {
```
тут тоже подумай какие константы можно вынести

kkkoltunov commented on src/main/java/edu/hw1/Task3.java:
```diff
@@ -0,0 +1,37 @@
+
+@SuppressWarnings("HideUtilityClassConstructor")
+public class Task3 {
+    public static boolean isNestable(int @NotNull [] fstMas, int[] secMas) {
```
обрати внимание на ворнинг, давай тоже избавимся от него

Skiftare replied:
> обрати внимание на ворнинг, давай тоже избавимся от него

Убрал бесполезный int ans;

kkkoltunov commented on src/main/java/edu/hw1/Task3.java:
```diff
@@ -0,0 +1,37 @@
+
+        for (int i = 0; i < fstMas.length; i++) {
+            if (i == 0) {
+                minFst = fstMas[i];
```
эту инициализацию можно вынести за пределы массива, только добавить проверку на длину массива перед этим

kkkoltunov commented on src/test/java/edu/hw1/Task1Test.java:
```diff
@@ -0,0 +1,43 @@
+    @DisplayName("Minutes_test")
+
+    @Test
+    void zero_zero() {
```
все тесты необходимо именовать в camelCase, так же как и методы и пременные

Skiftare committed:
Update Task1.java

Сделал через строки

kkkoltunov commented on src/test/java/edu/hw1/Task5Test.java:
```diff
@@ -0,0 +1,28 @@
+public class Task5Test {
+    @Test
+    void standart() {
+        assertEquals(true, Task5.isPalindromeDescendant(11211230));
```
можно воспользоваться `assertTrue`

Skiftare replied:
> можно воспользоваться `assertTrue`

Да, поправил. При этом это даже IDE предлагала, но я что-то проглядел этот момент

Skiftare committed:
Update Task2Test.java

Сделал тесты активными

kkkoltunov commented on src/main/java/edu/hw1/Task7.java:
```diff
@@ -0,0 +1,52 @@
+        }
+        Collections.reverse(binaryView);
+        int cycleMove = shift % len;
+        buf = 0;
```
можно убрать

Skiftare replied:
> можно убрать

Убрал, assertTrue/False переписал, camelCase сделал в тестах

Skiftare committed:
Update Task3.java

Убрал переменную

Skiftare committed:
Update Task2.java

Убрал magicNumber

kkkoltunov commented:
в целом неплохо, молодец! поправь пожалуйста комменты и я вернусь еще раз с проверкой (в каждом из тредов отпиши пожалуйста что ты согласен/не согласен и поправил/не поправил), из общего:
- старайся пользоваться возможностями ide, она подсвечивает много нюансов которые желательно исправлять (рекомендую Intelij Idea)
- в каждом классе делай автоформатирование кода, вот полезные хоткеи https://resources.jetbrains.com/storage/products/intellij-idea/docs/IntelliJIDEA_ReferenceCard.pdf
- старайся давать более осмысленное название переменным, не жалей символы, они бесплатные))

Skiftare committed:
Update Task3.java

Перенес инициализацию минимума-максимума, учёл пустые массивы

Skiftare committed:
Update Task1Test.java

camelCase

Skiftare committed:
Update Task7.java

Убрал cycleMove (штука, которая позволяла делать сдвиг больше, чем на длину числа в битовом представлении)

Skiftare committed:
Update Task5Test.java

Skiftare committed:
Update Task3Test.java

camelCase

Skiftare committed:
Update Task4Test.java

Skiftare committed:
Update Task6Test.java

camelCase

Skiftare committed:
Update Task7Test.java

camelCase

Skiftare committed:
Update Task8Test.java

camelCase + assertTrue/False

Skiftare committed:
Update Task6Test.java

0000 = 0. Поправил один тестик

Skiftare committed:
Update Task8Test.java

Импортировал assertTrue/False

Skiftare committed:
Update Task3Test.java

assertTrue/False

Skiftare committed:
Update Task1.java

Skiftare committed:
Update Task8.java

убрал magicNumber

Skiftare committed:
Update Task5.java

Skiftare committed:
Update Task4.java

Сделал больше буковок в переменных

Skiftare committed:
Update Task3.java

Skiftare committed:
Update Task3.java

Решил не изобретать велосипед и упростить. Добавил notNull, чтобы брать 0-элемент, сделал сортировку. Да, работает медленнее, зато читать приятнее.

Skiftare committed:
Update Task1Test.java

И тут я выяснил, что всё это время у меня было не 100% покрытие кода...

Skiftare committed:
Update Task1Test.java

Skiftare committed:
Update Task1.java

Skiftare committed:
Update Task1.java

Skiftare committed:
Update Task2.java

Притащил delLast из Task5

Skiftare commented:
- [ ] Лучше после ревью править файлы, и потом выкатывать один большой коммит, либо по файлам идти и сразу отправлять (файл-коммит, файл-коммит...)? 
- [ ] И насколько гитхаб нормально делает code coverage? Idea показывает 100%, например, Task1. И вот где эти 5% искать? Писать тесты для чего, для импорта библиотек, инициализаций? (Я загуглил. Гуглид как-то не так походу, инфы 0)

Skiftare committed:
Update Task7.java

Чуть повышена читабельность кода

Skiftare committed:
Update Task7Test.java

i want 100% cover!

Skiftare committed:
Update Task7.java

Skiftare committed:
Update Task2Test.java

kkkoltunov commented on src/main/java/edu/hw1/EvenArrayUtils.java:
```diff
@@ -0,0 +1,54 @@
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public final class EvenArrayUtils {
```
test

kkkoltunov commented on src/main/java/edu/hw1/Task1.java:
```diff
@@ -0,0 +1,36 @@
+    private static final Integer SIXTY_NUM = 60;
+
+    static int minutesToSeconds(@NotNull String s) {
+        int secs;
```
давай избегать именования переменных сокращенными словами, пиши плиз полностью `seconds`

kkkoltunov commented on src/main/java/edu/hw1/Task1.java:
```diff
@@ -0,0 +1,36 @@
+
+@SuppressWarnings("HideUtilityClassConstructor")
+class Task1 {
+    private static final Integer SIXTY_NUM = 60;
```
тут аналогично `SIXTY_NUMBER`, сейчас это не так важно, но учитывай что в будущем ты будешь работать в команде, твои сокомандники должны по названию переменной четко понимать что они делают 

поправь подобные нюансы пожалуйста в остальных частях кода

kkkoltunov commented on src/test/java/edu/hw1/Task1Test.java:
```diff
@@ -0,0 +1,50 @@
+public class Task1Test {
+
+    @DisplayName("Minutes_test")
+
```
лишний пробел, такие есть еще, пробегись пожалуйста по своему коду

kkkoltunov commented on src/test/java/edu/hw1/Task1Test.java:
```diff
@@ -0,0 +1,50 @@
+
+    }
+
+    @Test
```
давай ко всем тест классам методов напишем `@DisplayName` и описание, чтобы мне было проще проверять

kkkoltunov commented on src/test/java/edu/hw1/Task1Test.java:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    @Test
+    void a_lot_secs() {
```
тут тоже должен быть camelCase, проверь это пожалуйста **везде** в коде

kkkoltunov commented on src/test/java/edu/hw1/Task1Test.java:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    @Test
+    void standart() {
```
название теста должно говорить само за себя и желательно начинаться с префикса `test`, давай давать разумные имена `testThat...And...Returned...`

также нужно поправить **везде** в коде

Skiftare replied:
Оххх, понял... Ещё раз всё перепроверю. Невнимательность.

Skiftare replied:
> название теста должно говорить само за себя и желательно начинаться с префикса `test`, давай давать разумные имена `testThat...And...Returned...`
> 
> также нужно поправить **везде** в коде

Ладно, сейчас я начинаю понимать шутки про то, зачем Java-программистам 2 монитора. Чтобы переменные влезали.

kkkoltunov commented:
>  Лучше после ревью править файлы, и потом выкатывать один большой коммит, либо по файлам идти и сразу отправлять (файл-коммит, файл-коммит...)?

это дело вкуса, но кажется удобнее поправить все сразу и потом отправлять коммит (либо если устал закоммитить и продолжить дальше), не надо делать их так много)

>  И насколько гитхаб нормально делает code coverage? Idea показывает 100%, например, Task1. И вот где эти 5% искать? Писать тесты для чего, для импорта библиотек, инициализаций? (Я загуглил. Гуглид как-то не так походу, инфы 0)

честно говоря не знаю, тесты под импорт библиотек точно писать не надо, в целом у тебя код достаточно покрыт

Skiftare committed:
Update Task1Test.java

Пояснения к тестам

Skiftare committed:
Update Task1Test.java

camelCase

Skiftare committed:
Update Task3Test.java

DisplayName+ camelCase

Skiftare committed:
Update Task4Test.java

 DisplayName

Skiftare committed:
Update Task3Test.java

Skiftare committed:
Update Task4Test.java

Skiftare committed:
Update Task5Test.java

Skiftare committed:
Update Task6Test.java

Skiftare committed:
Update Task7Test.java

Skiftare committed:
Update Task8Test.java

Skiftare committed:
Update Task1.java

Skiftare committed:
Update Task5.java

Skiftare committed:
Push

Skiftare committed:
Нормальные наименования тестов

Skiftare committed:
Сделал код чуть более читабельным

Skiftare committed:
Исправление поломавшегося checstyle

Skiftare committed:
Update Main.java

Skiftare committed:
Исправление поломавшегося checstyle

Skiftare committed:
Исправление поломавшегося checstyle

Skiftare committed:
Update Task8.java

Skiftare committed:
Merge remote-tracking branch 'origin/hw1' into hw1

Skiftare committed:
Final

Skiftare committed:
Final

Skiftare committed:
Убрал enter

Skiftare committed:
Абсолютная дошлифовка

Skiftare committed:
Абсолютная дошлифовка тестов

Skiftare committed:
Дз2 - зачатки

kkkoltunov commented:
я вижу что у тебя уже появились в этой же ветке файлы связанные с hw2, ты должен сделать отдельную ветку с названием hw2, и в нее пушить изменения, этот мр мы вмерджим все равно когда закончатся правки

Skiftare committed:
Delete src/main/java/edu/hw2 directory

kkkoltunov commented:
в общем домашку я принимаю, ты молодец, хорошо поработал, давай в следующих домашках сохранять эту тенденцию и учитывать все правки и мои рекомендации, из замечаний оставлю: 
- следи за лишними пробелами, они все еще местами есть в конце или начале метода например
- следи за неймингом переменных, все еще есть переменные название которых не говорит само за себя например `funK`, `secN` и тд
- судя по coverage, покрыты не все возможные кейсы, старайся добивать полноценного покрытия

и пожалуйста давай комментарии на мои треды, что сделал, что не сделал, с чем согласен, какие есть сомнения и тд, возможно ты что-то считаешь лучше не делать, мне нужно твое мнение)

Skiftare commented:
> в общем домашку я принимаю, ты молодец, хорошо поработал, давай в следующих домашках сохранять эту тенденцию и учитывать все правки и мои рекомендации, из замечаний оставлю:
> 
> * следи за лишними пробелами, они все еще местами есть в конце или начале метода например
> * следи за неймингом переменных, все еще есть переменные название которых не говорит само за себя например `funK`, `secN` и тд
> * судя по coverage, покрыты не все возможные кейсы, старайся добивать полноценного покрытия
> 
> и пожалуйста давай комментарии на мои треды, что сделал, что не сделал, с чем согласен, какие есть сомнения и тд, возможно ты что-то считаешь лучше не делать, мне нужно твое мнение)

Безумие — это точное повторение одного и того же коммита. Раз за разом, в надежде на изменение. Это и есть безумие.

Если кратко по всему. После спортпрога очень тяжело писать много буковок, но это реально помогает - код становиться... Больше и читабельнее одновременно, хотя вот тесты я назвал так, что мне самому потребовались DisplayName, чтобы я понимал, что там твориться. secN я просто не придумал как поменять - второе n, чтобы IDE не ругалась, что я переданное использую. funK - так в дз написано, я нейминг не менял.

Skiftare commented:
> в общем домашку я принимаю, ты молодец, хорошо поработал, давай в следующих домашках сохранять эту тенденцию и учитывать все правки и мои рекомендации, из замечаний оставлю:
> 
> * следи за лишними пробелами, они все еще местами есть в конце или начале метода например
> * следи за неймингом переменных, все еще есть переменные название которых не говорит само за себя например `funK`, `secN` и тд
> * судя по coverage, покрыты не все возможные кейсы, старайся добивать полноценного покрытия
> 
> и пожалуйста давай комментарии на мои треды, что сделал, что не сделал, с чем согласен, какие есть сомнения и тд, возможно ты что-то считаешь лучше не делать, мне нужно твое мнение)

Пометьте пж ревью сделанным, чтобы я мог смержить. Конечно я и так и так это сделаю, но хочется чтоб красиво было
